### PR TITLE
Plugin installation parser

### PR DESCRIPTION
**Pull request by josenavas**
This PR allows to install/activate plugins from a config file generated by the plugins.

### CODE REVIEW

**Comment by antgonza** [comment]
@josenavas, there are actual errors here, what's the plan?

**Comment by josenavas** [comment]
I just restarted the build - I was not able to see the actual errors.

**Comment by antgonza** [comment]
flake8 fixed, tests failing:
```
======================================================================
ERROR: test_get_main (qiita_core.tests.test_configuration_manager.ConfigurationManagerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/biocore/qiita/qiita_core/tests/test_configuration_manager.py", line 37, in tearDown
    environ['QIITA_CONFIG_FP'] = self.old_conf_fp
  File "/home/travis/miniconda3/envs/env_name/lib/python2.7/os.py", line 473, in __setitem__
    putenv(key, item)
TypeError: putenv() argument 2 must be string, not None
```

**Comment by antgonza** [comment]
👍 , @wasade, could you review?

File: qiita_db/util.py [hunk 1]
```diff
@@ -43,7 +43,7 @@
 from __future__ import division
 from future.builtins import zip
 from future.utils import viewitems
-from random import choice
+from random import SystemRandom
```
Comment (wasade) [inline qiita_db/util.py]: why the change?

Comment (josenavas) [inline qiita_db/util.py]: There is a warning in the python documentation that said that random should not be used for security purposes, and it recommends the usage of SystemRandom. Since now I'm using the generate random string function to generate the client id / client secret pair for oauth2, I think is better to use SystemRandom.

**Comment by wasade** [comment]
thanks!
